src/RequestParser.cpp

```diff
@@ -1034,3 +1037,99 @@ bool RequestParser::isValidHostname(const std::string& hostname)
 	}
 	return hasAlpha;
 }
+
+/**
+ * @brief Checks for a single dot segment "/."
+ *
+ * Needs to check if iterator points to end, as derefencing end iterator is undefined behavior.
+ * @param iter Iterator to the current position.
+ * @param end End iterator of string.
+ * @return true It is a single dot segment.
+ * @return false It is not a single dot segment.
+ */
+static bool isSingleDot(const std::string::const_iterator& iter, const std::string::const_iterator& end)
+{
+	if (iter == end)
+		return false;
+	return *iter == '.' && (iter + 1 == end || *(iter + 1) == '/');
+}
+
+/**
+ * @brief Checks for a double dot segment "/.."
+ *
+ * Needs to check if iterator or iterator + 1 point to end, as derefencing end iterator is undefined behavior.
+ *
+ * @param iter Iterator to the current position.
+ * @param end End iterator of string.
+ * @return true It is a double dot segment.
+ * @return false It is not a double dot segment.
+ */
+static bool isDoubleDot(const std::string::const_iterator& iter, const std::string::const_iterator& end)
+{
+	if (iter == end || iter + 1 == end)
+		return false;
+	return *iter == '.' && *(iter + 1) == '.' && (iter + 2 == end || *(iter + 2) == '/');
+}
+
+/**
+ * @brief Removes last segment of a path denoted by '/'.
+ *
+ * @param output Reference to buffer containing path.
+ */
+static void removeLastSegment(std::string& output)
+{
+	if (!output.empty()) {
+		size_t lastSlash = output.find_last_of('/');
+		if (lastSlash != std::string::npos) {
+			output.erase(lastSlash);
+		} else {
+			output.clear();
+		}
+	}
+}
+
+/**
+ * @brief Removes dot segments "/." and "/.." from a path.
+ *
+ * Iterates through the string. If it finds a '/', increments and checks for
+ * - isSingleDot() = "/." - increment to ignore dot segment.
+ * - isDoubleDot() = "/.." - increment to ignore double dot segment and removeLastSegment() of output.
+ * - else - append a '/' to output buffer.
+ * If not a '/' append iter - 1 to buffer since we incremented in if-clause.
+ * If buffer is empty at the end append a '/' to it.
+ *
+ * This implements the algorithm as described in RFC 3986 Sect. 5.2.4.
+ * However it does not handle all described algorithm conditions
+ * - 2.A. buffer begins with "./" or "../" (note: '.' comes before '/')
+ * - 2.D. buffer consists only of '.' or '..'
+ * As requests need an absolute path (starting with '/') per RFC 9110 we can ignore these conditions.
+ *
+ * @sa https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4
+ * @param path Where to remove possible dot segments from.
+ * @return std::string String with dot segments removed.
+ */
+std::string RequestParser::removeDotSegments(const std::string& path)
+{
+	std::string output;
+	std::string::const_iterator iter = path.begin();
+
+	while (iter != path.end()) {
+		if (*iter++ == '/') {
```

Comment:
Would remove the post increment for better readability. Just add a iter++ in the line after. Then the output can be += *iter.

Reply:
Removed post increment, added iter++ in if and else clause